src/Microsoft.VisualStudio.Services.Agent/Constants.cs

```diff
@@ -87,6 +88,8 @@ public static class Agent
             public static readonly Architecture PlatformArchitecture = Architecture.Arm;
 #elif ARM64            
             public static readonly Architecture PlatformArchitecture = Architecture.Arm64;
+#elif RHEL6X64
+            public static readonly Architecture PlatformArchitecture = Architecture.Rhel6X64;
```

Comment:
base on Architecture, redhat 6 is still x64, not sure why we need to have this. :)

Reply:
I introduced a distinction between regular Linux x64 and RHEL6 x64, so jobs can take a decision based on what kind of agent they can demand (e.g. like I do with .vsts.ci.yml and .vsts.release.yml to build the RHEL6 packages).

As for why Architecture: it felt more natural to put the new distinction as part of the architecture (even though it feels a bit awkward for the constant to be in there), because after all, RHEL *is* Linux (so putting the constant in the OS enum would be flat-out wrong).

Now that I'm thinking this further: another option would be to introduce an additional agent enum -- something called `Flavor`. Regular Linux would become: `OS: Linux, Architecture: X64, Flavor: Generic`, and RHEL6 would be: `OS: Linux, Architecture: X64, Flavor: RHEL6`. This way, jobs can still tell apart RHEL6 and Generic Linux agents if they desire so. What do you think?

Reply:
Spoke with Ting. Because code-wise there were no changes required to build the RHEL6 agent package, I'm removing the special build constant, and specify the flavor as a custom capability.